### ISSUE TITLE
Porting models from v2.0.2 to current master 

### ISSUE THREAD

**Issue by AndRossi** [ModernMT/MMT#322]
In the current master version, and starting from commit 09981f7d35c366d5446e50a65d23c157ad90fabc, our neural models were made a bit different from earlier versions to achieve higher performances and translation quality.
  
If you have updated MMT to one of these recent commits, therefore, you need to upgrade your older engines in order to use them.

Don't worry: we have made this process entirely automatic :)
All you need to do is to run command from your mmt home:

```bash
extras/porting/convert_from_v2.0.2_to_v2.0.3 --engine-path <path to your engine folder>
```

For instance: 

```bash
extras/porting/convert_from_v2.0.2_to_v2.0.3 --engine-path engines/default
```
 
We will soon publish a new release `2.0.3` with these changes. As soon as that release is published, I will close this issue (and add the upgrading information on the Upgrade Engine page, of course).

If you have problems upgrading your engines, do not hesitate to contact us!

Thank you!

Andrea

**Comment by EtienneAb3d** [comment]
Can you explain what was improved ?
:)

**Comment by davidecaroselli** [comment]
@EtienneAb3d a detailed changelog will be included in the release as always. Short version: we have fixed a number of bugs (we will close pending issues in the next days), we implemented  a hot/cold strategy for Neural Engine in GPU RAM, we changed the termination condition after a wide range of experiments on different language pairs.

**Comment by EtienneAb3d** [comment]
Can't wait to get technical details...
:D

**Comment by mzeidhassan** [comment]
Hi @AndRossi ,

I am a bit confused. I did the following:

> git pull origin master

but running the command above throws this error:

`From https://github.com/ModernMT/MMT
 * branch            master     -> FETCH_HEAD
Updating 52116af..1573f5b
error: Your local changes to the following files would be overwritten by merge:
        src/core/src/main/java/eu/modernmt/facade/TrainingFacade.java
Please, commit your changes or stash them before you can merge.
Aborting`

Any idea?

**Comment by davidecaroselli** [comment]
Hi @mzeidhassan 

you have some changes on your code, that's why git is complaining. For a clean build I always recommend this procedure:
```
git checkout -- .
git clean -fd .

rm -rf build/*
git pull origin master

cd vendor
./compile
cd ../src
mvn clean install

**Comment by mzeidhassan** [comment]
Perfect, @davidecaroselli ! This fixed the issue and the engine has been ported successfully.

`Congratulations! Your 2.0.2 engine was successfully ported to version 2.0.3`  :-)

**Comment by AndRossi** [comment]
Hi everyone, 
We have released new MMT version `2.1` (you will find it in our `master` branch as well as in our releases page) so, as I said, I'm closing this issue now.

Contrarily to our first decision, the new version is called `2.1` instead of `2.0.3`, so the script has been renamed `extras/porting/convert_from_v2.0.2_to_v2.1`. Apart from that, the behaviour is exactly the same.

I have already updated the [upgrade page](https://github.com/ModernMT/MMT/wiki/Upgrading-MMT) in our Wiki.

Thanks for your support!

Andrea
